redi-connect-front/src/pages/app/dashboard/mentee/AvailableMentorListing.tsx

```diff
@@ -1,114 +1,191 @@
+
+// const addMatchCount = (
+//   mentors: RedProfile[],
+//   categories: string[],
+//   languages: string[]
+// ): MentorCatCount[] =>
+//   mentors.map(mentor =>
+//     Object.assign(mentor, {
+//       categoryMatchCount: intersection(categories, mentor.categories).length,
+//       languageMatchCount: intersection(languages, mentor.languages).length
+//     })
+//   )
+
+interface FilterTagProps {
+  id: string
+  label: string
+  onClickHandler: (item: string) => void
+}
+
+const FilterTag = ({ id, label, onClickHandler }: FilterTagProps) => <Tag
+  key={id}
+  categoryId={id}
+  size="medium"
+  rounded
+  textWeight="bold"
+>
+  {label}
+  <Icon icon="cancel" onClick={() => {
+    console.log(id)
+    onClickHandler(id)
+  }} className="active-filters__remove" />
+</Tag>
 
 export const AvailableMentorListing = (props: any) => {
   const { Loading, isLoading, setLoading } = useLoading()
-  const [_mentors, setMentors] = useState<RedProfile[]>([])
+  const [mentors, setMentors] = useState<RedProfile[]>([])
   const currentUserCategories = getRedProfile().categories
-  const [activeCategories, { toggle }] = useList(currentUserCategories)
+  const [activeCategories, { toggle: toggleCategories, clear: clearCategories }] = useList(currentUserCategories)
+  const [activeLanguages, { toggle: toggleLanguages, clear: clearLanguages }] = useList<any>([])
 
-  const mentorsFiltered = _mentors.filter(
-    m => m.currentFreeMenteeSpots > 0 && m.userActivated
-  )
+  // const mentorsFiltered = _mentors.filter(
+  //   m => m.currentFreeMenteeSpots > 0 && m.userActivated
+  // )
 
-  const mentorsWhoHaveSpotsButAreNotActivatedCount = _mentors.filter(
-    m => m.userActivated === false && m.currentFreeMenteeSpots > 0
-  ).length
+  // const mentorsWhoHaveSpotsButAreNotActivatedCount = _mentors.filter(
+  //   m => m.userActivated === false && m.currentFreeMenteeSpots > 0
+  // ).length
 
-  const mentors = addCategoryMatchCount(mentorsFiltered, activeCategories)
+  // const mentors = addMatchCount(mentorsFiltered, activeCategories, activeLanguages)
 
-  const mentorsWithSharedCategories = mentors
-    .filter(m => m.categoryMatchCount > 0)
-    .sort((a, b) => b.categoryMatchCount - a.categoryMatchCount)
-  const mentorsWithoutSharedCategories = mentors.filter(
-    m => m.categoryMatchCount === 0
-  )
+  // const mentorsWithSharedCategories = mentors
+  //   .filter(m => m.categoryMatchCount > 0)
+  //   .sort((a, b) => b.categoryMatchCount - a.categoryMatchCount)
+
+  // const mentorsWithoutSharedCategories = mentors.filter(
+  //   m => m.categoryMatchCount === 0 && m.languageMatchCount === 0
+  // )
+
+  // const filterCategories = Array.from(
+  //   new Set(
+  //     mentors
+  //       .map(mentor => mentor.categories || [])
+  //       .flat()
+  //       .sort()
+  //   )
+  // ).map(categorie => ({
+  //   value: categorie,
+  //   label: categoriesIdToLabelMap[categorie]
+  // }))
+
+  const filterLanguages = Array.from(
```

Comment:
Can probably be simplified by using lodash' `uniq`

Reply:
Also, though I didn't see it, I think this will lead to languages disappearing from the filter list if a user does a very narrow filtering. The list of filter languages should be generated from the very first fetch, and nothing else.

Reply:
Thas is true, we are still discussing about how to handle the list, there are multiple possibilities. This one include updating the filter based on the current list. Still an open topic so I will, for now ignore this comment.